### PR TITLE
Maps

### PR DESCRIPTION
**Pull request by theandrewmo**
### Description:
This pull request adds a maps feature to the brewery details page, leveraging the Google Maps API to display a map showing the location of the brewery. This enhancement aims to improve the user experience by providing visual context and helping users easily locate the brewery they are interested in.

### Changes:

-Integrated the Google Maps API library into the project.
-Created a new section on the brewery details page to display the map.
-Extracted the latitude and longitude coordinates of the brewery
-Implemented JavaScript code to initialize and display the map with a marker at the brewery location.
-Customized the map appearance to match the overall design of the application.

### Testing:
To ensure the proper functioning of the maps feature, the following testing steps were performed:

-Manually tested the brewery details page with different breweries, verifying that the map correctly shows the location.
-Checked for responsiveness and compatibility.
-Tested the map marker's accuracy by comparing the displayed location with known coordinates.

### Dependencies:
This feature depends on the following:

The Google Maps API library
The API storing the brewery details (https://api.openbrewerydb.org/v1/), including the latitude and longitude coordinates.

### Screenshots / Demo:

Screen Shot 2023-05-23 at 11 26 55 AM

### Additional Notes:

The Google Maps API key is securely stored in the environment variables to ensure privacy and restrict unauthorized usage.
The design of the map section is kept minimalistic to maintain focus on the brewery information.
Future improvements could include adding additional map features, such as directions, to enhance the user experience.

### CODE REVIEW

File: static/app.js [hunk 1]
```diff
@@ -31,17 +36,124 @@ $(function() {
...
+    /** helper function to process brewery API response data
+     * takes response data and dataType as parameters, turns response data into an results object to be rendered
+     * dataType is in regards to autocomplete or normal results
+     */
 
-    if (!sessionStorage.getItem('latitude') && !sessionStorage.getItem('longitude') && 'geolocation' in navigator) {
-        getLocation()
+    function processBreweries(data, dataType){
+        const breweriesResults = Object.values(data).map((brewery) => {
+            return createBrewery(brewery, dataType)
```
Comment (akhayoon) [inline static/app.js]: nit: you don't need the return here. you can just `.map(brewery => createBrewery...`

File: static/app.js [hunk 2]
```diff
@@ -75,113 +189,161 @@ $(function() {
...
+
+    async function initMap() {
+    const { Map } = await google.maps.importLibrary("maps");
+    let location = {lat: parseFloat($('#latitude').val()), lng: parseFloat($('#longitude').val())}
+    if (location.lat && location.lng) {
+        map = new Map($('#map')[0], {
+            center: location,
+            zoom: 12,
+        });
+        let marker = new google.maps.Marker({position:location, map:map})
```
Comment (akhayoon) [inline static/app.js]: Use `const` if not changing assignment. And use shorthand with key values like `map: map` to just ` {map}`

File: static/app.js [hunk 3]
```diff
@@ -75,113 +189,161 @@ $(function() {
...
+    else {
+        $('#map').addClass('d-flex flex-column border border-primary border-2 align-items-center justify-content-center')
+                .html(`<p>Unable to render map</p>
+                       <p>Coordinates not available</p>
+                    `)
+    }}
+
+    /** conditional that calls initMap when the window location is the brewery-detail page */
+
+    if (window.location.href.match(/\/breweries\/[a-f0-9-]+$/)) {
```
Comment (akhayoon) [inline static/app.js]: maybe turn this into a utility function for searching up route destinations if you do this more than once in the future

File: templates/base.html [hunk 4]
```diff
@@ -9,6 +9,14 @@
...
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css" integrity="sha512-iecdLmaskl7CVkqkXNQ/ZH/XLlvWZOJyj7Yy7tcenmpD1ypASozpmT/E0iPtmFIB46ZmdtAc9eNBvH0H/ZpiBw==" crossorigin="anonymous" referrerpolicy="no-referrer" />
     <link rel="stylesheet" href="/static/app.css">
+    <script>
+        (g=>{var h,a,k,p="The Google Maps JavaScript API",c="google",l="importLibrary",q="__ib__",m=document,b=window;b=b[c]||(b[c]={});var d=b.maps||(b.maps={}),r=new Set,e=new URLSearchParams,u=()=>h||(h=new Promise(async(f,n)=>{await (a=m.createElement("script"));e.set("libraries",[...r]+"");for(k in g)e.set(k.replace(/[A-Z]/g,t=>"_"+t[0].toLowerCase()),g[k]);e.set("callback",c+".maps."+q);a.src=`https://maps.${c}apis.com/maps/api/js?`+e;d[q]=f;a.onerror=()=>h=n(Error(p+" could not load."));a.nonce=m.querySelector("script[nonce]")?.nonce||"";m.head.append(a)}));d[l]?console.warn(p+" only loads once. Ignoring:",g):d[l]=(f,...n)=>r.add(f)&&u().then(()=>d[l](f,...n))})({
+          key: "{{google_maps_api_key}}",
```
Comment (akhayoon) [inline templates/base.html]: Great work, hiding this

File: static/app.js
Comment (akhayoon) [inline static/app.js:1]: Consider splitting this file into multiple domains files and import using script tags

**Review by akhayoon — APPROVED** [review]

File: static/app.js (hunk 1 above)
Comment (theandrewmo) [reply static/app.js]: i noticed that this doesn't work without the return statement so leaving there for now.

File: static/app.js (hunk 2 above)
Comment (theandrewmo) [reply static/app.js]: done!

File: static/app.js (hunk 3 above)
Comment (theandrewmo) [reply static/app.js]: noted

File: templates/base.html (hunk 4 above)
Comment (theandrewmo) [reply templates/base.html]: thanks!

File: static/app.js
Comment (theandrewmo) [reply static/app.js:1]: done!

**Pull request merged**